### PR TITLE
Add tests for equals, hashCode and compareTo in LineColumn. #1088

### PR DESCRIPTION
**Pull request by mkordas**

### CODE REVIEW

**Comment by romani** [comment]
I always in favour of not overriding hashcode/equals method in java classes and usage of standalone compare methods, am I only ok for with overriding for immutable classes.
 
I originally keep a copareTo and added hashcode/equals as it located in class that is in API, so not only our code have to be considered on decision to change class hierarchy.

Please think one more time about this update and explain your motivations.

**Comment by mkordas** [comment]
My motivation was simple - these methods are not used in Checkstyle code at all and they had 0% code coverage. But as you've noticed this is API - I'll need to check whether other projects need these methods.

**Comment by romani** [comment]
As we discussed at another issue, lets keep that methods for now as it is api and we do not know all usages.

**Comment by romani** [comment]
@mkordas , lets finish that PR.

**Comment by mkordas** [comment]
@romani, I'm working on it.

File: src/test/java/com/puppycrawl/tools/checkstyle/api/LineColumnTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public void testCompareToFirstLarger() {
+        LineColumn lineColumn = new LineColumn(0, 0);
+
+        assertEquals(1, new LineColumn(1, 0).compareTo(lineColumn));
+        assertEquals(1, new LineColumn(0, 1).compareTo(lineColumn));
+    }
+
+    @Test
+    public void testCompareToFirstSmaller() {
+        Comparable<LineColumn> lineColumn = new LineColumn(0, 0);
```
Comment (mkordas) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/LineColumnTest.java]: To achieve 100% code coverage method had to be invoked also through interface to cover bridge method: http://codeidol.com/java/javagenerics/Comparison-and-Bounds/Bridges/

**Comment by romani** [comment]
merged as FF

**Pull request closed without merging**